lib/puppet/ssl/state_machine.rb

```diff
@@ -422,6 +438,7 @@ def with_lock
         @lockfile.unlock
       end
     else
+      raise Puppet::LockError, _('Another puppet instance is already running; --waitforlock flag used, waiting for running instance to finish') if Puppet[:waitforlock]
```

Comment:
I suppose it should be safe to always raise `Puppet::LockError`, here since this code runs when the file cannot be locked, and the current handling of `Puppet::LockError` could check if the rameter is set as is already checking it's value

Reply:
Seems there are some tests regarding StateMachine that require with_lock to raise Puppet::Error. Instead of modifying the tests, maybe it's ok to have both raises here so that errors raised remain consistent in calls without waitforlock flag.

Reply:
If we always raise `Puppet::LockError` here and delete the line below, then the rescue handling above can differentiate between the 3 cases. In the first 2 cases (waitforlock is 0 or the deadline was exceeded), then raise `Puppet::Error` like we were doing before, so API remains consistent. I'd expect https://github.com/puppetlabs/puppet/blob/master/spec/unit/ssl/state_machine_spec.rb#L231-L236 to not need changing.